src/main/java/com/epam/ta/reportportal/dao/IntegrationRepositoryCustomImpl.java

```diff
@@ -38,4 +46,67 @@
 
 		return dsl.fetch(QueryBuilder.newBuilder(filter).build()).map(INTEGRATION_MAPPER);
 	}
+
+	private final EntityManager entityManager;
+
+	@Autowired
+	public IntegrationRepositoryCustomImpl(EntityManager entityManager) {
+		this.entityManager = entityManager;
+	}
+
+	@Transactional
+	@Override
+	public void deleteSettings(Integration integration) {
+		entityManager.remove(integration);
+	}
+
+	@Transactional
+	@Override
+	public void updateLdap(LdapConfig ldapConfig) {
+		entityManager.merge(ldapConfig);
+
+	}
+
+	@Transactional
+	@Override
+	public void updateActiveDirectory(ActiveDirectoryConfig adConfig) {
+		entityManager.merge(adConfig);
+	}
+
+	@Override
+	public Optional<LdapConfig> findLdap(boolean enabled) {
+		TypedQuery<LdapConfig> ldapConfigTypedQuery = entityManager.createQuery(
+				"SELECT l FROM LdapConfig l JOIN Integration i ON l.id = i.id WHERE i.id = :id AND i.enabled = :enabled",
+				LdapConfig.class
+		);
+
+		ldapConfigTypedQuery.setParameter("id", 1L);    //should be custom id, that was received from user
+		ldapConfigTypedQuery.setParameter("enabled", enabled);
+
+		return ldapConfigTypedQuery.getResultList().stream().findFirst();
+	}
+
+	@Override
+	public Optional<ActiveDirectoryConfig> findActiveDirectory(boolean enabled) {
+		TypedQuery<ActiveDirectoryConfig> activeDirectoryConfigTypedQuery = entityManager.createQuery(
+				"SELECT a FROM ActiveDirectoryConfig a JOIN Integration i ON a.id = i.id WHERE i.id = :id AND i.enabled = :enabled",
+				ActiveDirectoryConfig.class
+		);
+
+		activeDirectoryConfigTypedQuery.setParameter("id", 1L);   //should be custom id, that was received from user
```

Comment:
do we need to do smth with this?

Reply:
Id is hardcoded because we still don't know the structure of the client request. 
The simpliest way of retrieving auth configuration from database was chosen for ldap functionality testing.